sequence/src/main/java/com/vitreoussoftware/bioinformatics/sequence/encoding/ExpandedIupacEncodingScheme.java

```diff
@@ -293,6 +305,39 @@ public BasePair toBasePair(byte nucleotide) throws InvalidDnaFormatException {
 
     @Override
     public BasePair fromCharacter(Character character) {
-        return AcceptUnknownDnaEncodingScheme.create(character);
+        return ExpandedIupacEncodingScheme.create(character);
+    }
+
+    @Override
+    public BasePair complement(BasePair basePair) {
+        if (!basePair.getEncodingScheme().equals(this)) {
+            return complement(fromCharacter(basePair.toChar()));
+        }
+
+        byte nucleotide = basePair.getValue();
+        if ((nucleotide & AMBIGUITY) == 0) {
+            if ((nucleotide & SHIFT_RIGHT) != 0)
+                return toBasePair((byte) (nucleotide >>> 1));
+            return toBasePair((byte) ((nucleotide & LOWER_ORDER) << 1));
+        }
+        else if ((nucleotide & REFLECTED) != 0) {
+            return basePair;
```

Comment:
It is not generically correct, no. However once you each that point in the code the only remaining 'reflected' options are their own complement (N - any, X - there is a gap)